core/src/main/scala/scalaz/EitherT.scala

```diff
@@ -287,6 +287,39 @@ object EitherT extends EitherTInstances {
 
 }
 
+object EitherTNondeterminism {
+  implicit def eitherTNondeterminism[F[_], L](implicit F0: Nondeterminism[F]): Nondeterminism[EitherT[F, L, ?]] =
+    new Nondeterminism[EitherT[F, L, ?]] {
+      val F = Nondeterminism[F]
+
+      override def chooseAny[A](
+                                 head: DisjunctionT[F, L, A],
+                                 tail: Seq[DisjunctionT[F, L, A]]): DisjunctionT[F, L, (A, Seq[DisjunctionT[F, L, A]])] =
+        new EitherT(F.map(F.chooseAny(head.run, tail map (_.run))) {
+          case (a, residuals) =>
+            a.map((_, residuals.map(new EitherT(_))))
+        })
+
+      override def bind[A, B](fa: DisjunctionT[F, L, A])(f: (A) => DisjunctionT[F, L, B]): DisjunctionT[F, L, B] =
+        fa flatMap f
+
+      override def point[A](a: => A): DisjunctionT[F, L, A] = EitherT.eitherTMonad[F, L].point(a)
+
+      override def reduceUnordered[A, M](fs: Seq[DisjunctionT[F, L, A]])(
+        implicit R: Reducer[A, M]): DisjunctionT[F, L, M] = {
+        import R.monoid
+        val AE = Apply[L \/ ?]
+        val RR: Reducer[L \/ A, L \/ M] =
+          Reducer[L \/ A, L \/ M](
+            _.map(R.unit),
+            c => AE.apply2(c, _)(R.cons(_, _)),
+            m => AE.apply2(m, _)(R.snoc(_, _))
+          )(Monoid.liftMonoid[L \/ ?, M])
+        new EitherT(F.reduceUnordered(fs.map(_.run))(RR))
+      }
```

Comment:
Why have you overridden the default implementation of `reduceUnordered`? The default implementation seems to have some desirable short-circuit behavior for `EitherT` that your override does not preserve.